### PR TITLE
Introduce Enhanced Consent plugin

### PR DESCRIPTION
**Pull request by Jack-Keene**
Introduces an enhanced consent plugin for tracking consent events and CMP load time events.

Details of the API can be found here:
https://www.notion.so/JS-tracker-enhanced-consent-API-7aa67bb253e14c6cab01090f120556be?d=390079798b994ae0a8374c9bd9f7b621

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

**Review by igneel64 — APPROVED** [review]
Great 🚀

**Review by paulboocock — COMMENTED** [review]
Looks good! I think it needs adding into here: https://github.com/snowplow/snowplow-javascript-tracker/blob/master/trackers/javascript-tracker/src/features.ts so it can be easily included in a custom build of sp.js (or sp.lite.js).

I do wonder if we should include this by default, it doesn't seem too big? And we could remove the old gdpr tracking in v4 to make up for it.

File: plugins/browser-plugin-enhanced-consent/package.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "@snowplow/browser-plugin-enhanced-consent",
+  "version": "3.6.0",
```
Comment (Jack-Keene) [inline plugins/browser-plugin-enhanced-consent/package.json:3]:
```suggestion
  "version": "3.7.0",
```

**Pull request merged**